students/aravichander/lesson09/Mailroom_class.py

```diff
@@ -73,54 +110,57 @@ def letterofthanks(donorname,totaldonated):
 	f.close()
 	print("\nLetter of thanks has been saved to hard disk as text file under "+str(filename))
 
-def sumdonor(thankdonor):
-	return sum(Donors[thankdonor])
+# def sumdonor(thankdonor): #redundant - refactor to remove this
+# 	return sum(Donors[thankdonor])
 
-def searchdonors(thankdonor):
+def searchdonors(thankdonor,db):
 		try:
-			print(Donors[thankdonor])
-			totaldonated = sumdonor(thankdonor)
+			#print(db[thankdonor])
+			donor = db.get_donor(thankdonor)
+			totaldonated = donor.sum_donation
 			letterofthanks(thankdonor,totaldonated)
 		except KeyError:
 			print("\nThat donor doesn't exist")
 
 
-def newdonation():
+def newdonation(db):
 	newdonor = input("Input name of new donor\n")
 	while len(str.strip(newdonor)) == 0:
 		newdonor = input("Invalid input, enter name of new donor\n")
 	newdonor = str.strip(newdonor)
-	Donors[newdonor] = []
+	try:
+		newdonor_object = db.get_donor(newdonor)
+	except:	
+		newdonor_object = Donor(newdonor)
 	donationsdone = False
-	newdonations = []
 	while donationsdone is False:	
 		donation = input("Input donations or hit enter when done\n")
 		if donation == "":
 			donationsdone = True 
 		else:
 			try:
-				newdonations.append(float(donation))
+				newdonor_object.add_donation(float(donation))
 			except ValueError:
 				print("Enter number or hit enter - no letters!")
 	#print("Donations is/are {}".format(newdonations))
-	Donors[newdonor] = newdonations
-	printdonors()
+	db.add_donor(newdonor_object)
+	db.print_Db()
 
-def thanks():
+def thanks(db):
 	thankdonor = input("\nPlease enter full name of donor you'd like to thank or list to view donors\n")
 	if thankdonor == "list":
-		printdonors()
-		thanks()
+		db.print_Db()
+		thanks(db)
```

Comment:
careful with recursion in an interactive loop -- this should be a while loop, so it can go on forever if need be.